### ISSUE TITLE
Tkinter checks in python easyblock break bundles that install the python interpreter

### ISSUE THREAD

**Issue by damianam** [hpcugent/easybuild-easyblocks#1181]
The problem lies in these lines:

https://github.com/hpcugent/easybuild-easyblocks/blame/master/easybuild/easyblocks/p/python.py#L152
https://github.com/hpcugent/easybuild-easyblocks/blame/master/easybuild/easyblocks/p/python.py#L194

`self.version.split('.')[:2]` works fine as long as the module version is the python version. But if you have an easyconfig that installs the python interpreter and some extra packages, with a version that is not a python version, it breaks, whereas before it worked. The use case is having `Python` in `GCCcore` and `SciPy-Stack` at the top level, including in `SciPy-Stack` an additional python interpreter to use the optimized exponential and trigonometrical functions from `libimf`

**Comment by boegel** [comment]
@damianam Hmm, can you show an easyconfig file like this?

Aren't you also seeing a weird symlink `python[0-9]*` being installed (see `install_step`)?

Any suggestions on how to fix this?

**Comment by damianam** [comment]
Sure thing: 
[SciPy-Stack-2017a-gmvolfc-2017a-GDR-Python-2.7.13.eb.txt](https://github.com/hpcugent/easybuild-easyblocks/files/1068014/SciPy-Stack-2017a-gmvolfc-2017a-GDR-Python-2.7.13.eb.txt)

About suggestions. What about replacing this

`pyver = 'python' + '.'.join(self.version.split('.')[:2])`

 with

`(pyver, _) = run_cmd("python -c 'import sys; print(".".join(map(str, sys.version_info[:2])))'", log_all=True, log_output=verbose)`
?

**Comment by boegel** [comment]
@damianam I've taken a closer look at this, and I feel you're abusing (that's a strong word, but you get the idea) the `EB_Python` easyblock for something that actually requires a custom easyblock.

Relying on `self.version` in `EB_Python` to determine the version of the Python being installed is very natural, collecting it via `run_cmd` is counter-intuitive (and hence more confusion/mistakes will occur in the future with it).

What's worse, even though obtaining the Python version via `run_cmd` would indeed work in the sanity check (since a temporary version of the module that is generated then will be loaded), it won't work in other places of the `EB_Python` easyblock where `self.version` is currently being used, i.e. in `install_step` where a symlink `bin/python` that points to `bin/python<pyver>` is being put in place (check your `bin` directly in your `SciPi-Stack` installation, you'll have a funny looking/wrong `bin/python` symlink in there, I think?).
The `<pyver>` part can only be derived from `self.version` in `install_step`, not by running a `python` command (at least not in a general sense).

I think we have two options:

* enhance the `EB_Python` easyblock so you can redefine `self.version` early on via a custom easyconfig parameter; for your use case,  i.e. building Python while having another Python loaded as a dependency, it should be defined to be equal to `$EBVERSIONPYTHON` (obtained via `get_software_version('Python')

* create a dedicated (generic) easyblock `PythonStack` (or whatever) to install `SciPy-Stack`, that derives from `EB_Python`, and takes care of all the magic to build a stack of Python packages on top of an existing Python while also building a custom Python interpreter (which is a bit of a weird situation by itself); the advantage of that approach is that you can put checks in place to make sure this keeps working, and it would allow you to clean up the `SciPy-Stack` easyconfig file

I'd strongly prefer the latter.

Long story short: we can fix the (unintended) breakage in `EB_Python` for your use case, but it may happen again in the future since it's quite natural to use `self.version` in the Python easyblock and assume it specifies the version of the Python being installed; that's sort of why the breakage occurred in the first place, I certainly wasn't (fully) aware that the `EB_Python` easyblock was being (ab)used in this way.

Let me know what you think...

**Comment by boegel** [comment]
There's actually an easier way out too: version `SciPy-stack` so that `self.version` can still be used, for example use `2.7.13-2017a` as version (and drop the `-Python-2.7.13` `versionsuffix`).

**Comment by damianam** [comment]
@boegel no need to sugar coat it, what I was doing totally deserve the word "abuse" ;-). But it worked just fine so I rolled with it. I agree that having a derived `PythonStack` easyblock does indeed make sense. I'll look into it.

ps: I do not have any wrong symlink in `$EBROOTSCIPYMINSTACK/bin`. I have one that is funny looking:

```
~]$ ls -l python*
lrwxrwxrwx 1 $USER $GROUP     7 Apr  4 11:56 python -> python2
lrwxrwxrwx 1 $USER $GROUP     9 Apr  4 11:56 python2 -> python2.7
-rwxr-xr-x 1 $USER $GROUP 11544 Apr  4 11:56 python2.7
...
```

but it is the same in the the normal python.

**Comment by boegel** [comment]
@damianam The `EB_Python` easyblock will only put a symlink for `python` in place if `bin/python` doesn't exist yet, so that explains the lack of a funny looking symlink.

**Comment by damianam** [comment]
As discussed, closing this issue, pending of a new easyblock.
